src/Schema.ts

```diff
@@ -277,6 +277,61 @@ class Schema<
 		return [...positions];
 	}
 
+	/** Transform the paths to ordinal positions. Returning a '.' delimited string. Ex. "1.2.3"  */
+	public transformPathsToOrdinalPositions(paths: string[]): string[] {
```

Comment:
This method feels weird for several reasons:
1. You are accepting a `string[]` and returning a `string[]` but the only use case calls for passing a single value in and getting a single value back.
2. The logic in the `else if` seems improper. For the projection use case, if a parent key was supplied then it needed to also produce paths for all children. In this use case, the expectation would be that a full keypath is supplied and not partial keypaths.
3. The `paths` isn't really a `string[]`. It's specifically the keys from the flattened document structure.
4. The return type is specifically `${number} | ${number}.${number} | ${number}.${number}.${number}`.
5. If the path isn't found then returning empty string or potentially an empty list seems improper. That would then set the `path` that gets passed to basic as an empty string or undefined value and then cause problems on the basic side.

Reply:
> 3. The paths isn't really a string[]. It's specifically the keys from the flattened document structure.
I'm trying to get this to work, but I'm struggling with how can I pass the Schema to `FlattenDocument` or `InferSchemaPath` when I'm within Schema. I've tried passing `this` and `typeof this` as the type parameter, but that wasn't working.

Reply:
This worked for me:
```
	/** Transform the paths to ordinal positions. Returning a '.' delimited string. Ex. "1.2.3"  */
	public transformPathsToOrdinalPositions(
		paths: Extract<
			keyof FlattenDocument<typeof this, Schema | SchemaDefinition | SchemaTypeDefinitionNumber>,
			string
		>[],
	): string[] {
	```
Although I would recommend making an alias for that and using it in both places it would be needed.

Reply:
I was getting infinite type instantiation as well

Reply:
```
	public transformPathsToOrdinalPositions(
		path: InferSchemaPaths<Schema<TSchemaDefinition>>,
	): string {
	```
	
	Since that method is on `Schema` it doesn't require the same restrictions to the paths that the increment method requires. So I think this will work.

Reply:
That works but then the type of the paths is a union of  `string | number | symbol`. I'm casting the value to `String`, but I'm not 100% sure that's accurate. I'm not entirely sure why it's a union of the 3 types either.

Reply:
Interesting I would not have thought to just extract the string type from that. Why does that work? It makes sense what Extract does, but why is it valid/the right thing to do, to just ignore the other types?

I wrap `InferSchemaPaths` as such: `Extract<InferSchemaPaths<Schema<TSchemaDefinition>>, string>` and that appears to work within `transformPathsToOrdinalPositions`, but I end up with `never` when using `transformPathsToOrdinalPositions`.
![image](https://github.com/user-attachments/assets/a24a6377-704b-4755-a557-194bcd8b6f53)

Reply:
I'm thinking it has to do with the `TDictionaries` as it defaults to `Record<never, never>`

Reply:
but passing `TDictionaries` along does not help.